### ISSUE TITLE
make text marking slightly smoother

### ISSUE THREAD

**Issue by AndiBraimllari** [OpenCovenant/quill#92]
When a typo is currently found, the styles that are applied to it feel quite sudden, perhaps they can be done slightly smoother (maybe through a quick transition or so).

Please note that this issue is only about making the feel of the styling a bit more natural/smoother, and no logical change should be made to the markings for now.

**Comment by vaqueraoscar0** [comment]
@AndiBraimllari are you referring to these boxes here on the side. If so would a simple fade-in effect be enough
<img width="1311" alt="Screenshot 2023-09-27 at 2 21 00 PM" src="https://github.com/OpenCovenant/quill/assets/57508905/28d6215d-c298-428a-a16f-153f46e68647">

**Comment by AndiBraimllari** [comment]
I'm referring to the markings in the editor (left hand side), e.g. 
![Screenshot from 2023-09-27 23-43-23](https://github.com/OpenCovenant/quill/assets/11479070/aa02cf68-41a2-4efb-b168-a50b0e5e0fb9)
After a request is made, the markings in the editor are shown right away, maybe there's a way of doing it a bit smoothly.

**Comment by vaqueraoscar0** [comment]
oh, i understand now. The only thing i can think of right now, is adding some animation to it like a color fill that pre-fills from left to right, or maybe a fade in. 

https://github.com/OpenCovenant/quill/assets/57508905/6cc97b55-46d6-45f0-b52b-4768b751183b

**Comment by AndiBraimllari** [comment]
Yup, I think something like that (perhaps the fade-in can be a bit faster) would be pretty cool.
What do you make of it, my reasoning is that the way the markings are applied to the text should feel natural (which is why I like some sort of brief animation to it).
If you'd like, you can add these changes in a PR and we can further discuss them there.

**Comment by vaqueraoscar0** [comment]
@AndiBraimllari Sounds good, i'll go ahead and start the process and make the changes for it.

**Comment by vaqueraoscar0** [comment]
@AndiBraimllari I agree, there’s nothing wrong with how it was before, but you’re right, adding some sort of secondary effect makes the entire experience feel different.
